### ISSUE TITLE
Broken handling of inheritance after "Optimize zero-argument methods"

### ISSUE THREAD

**Issue by pustomytnyk** [panorama-ed/memo_wise#228]
Seem that after d768e1091ae112531c4bf34534f43aa1f37c9761 inheritance is not handled properly, because used indices are mixed up between classes.

```ruby
require 'memo_wise'

class Parent
  prepend MemoWise

  def bar
    'bar'
  end
  memo_wise :bar
end

class Child < Parent
  def foo
    'foo'
  end
  memo_wise :foo
end

child = Child.new
pp child.foo # initialize Child sentinels
pp child.bar # expected 'bar', got 'foo' because Child sentinels used instead of Parent
```

**Comment by ms-ati** [comment]
😃 thank you for catching this and reporting so clearly @pustomytnyk !

**Comment by JacobEvelyn** [comment]
Thanks for reporting this, @pustomytnyk! I think this may actually be an issue for methods that take arguments as well. @jemmaissroff and I are working on a fix.

**Comment by JacobEvelyn** [comment]
Update: we think we have a working solution and I can make a PR tomorrow! Thanks for your patience @pustomytnyk !

**Comment by JacobEvelyn** [comment]
@pustomytnyk sorry this took a bit longer than anticipated—Ruby metaprogramming can be tricky! The fix for this is now released in v1.3.0. If you experience any other issues please let us know! ❤️
